frontend/src/actions/authActions.ts

```diff
@@ -0,0 +1,39 @@
+import { takeLatest, call, put } from 'redux-saga/effects';
+import { post } from '../apis';
+import { authTypes, AuthAttempt, LinkRequest } from '../types/authTypes';
+
+import { push } from 'react-router-redux';
+
+
+
+export function* rootSaga() {
+    yield takeLatest(authTypes.AUTH_ATTEMPT, auth);
+    yield takeLatest(authTypes.LINK_REQUEST, requestLink);
+}
+
+//sagas
+
+function setAuthToken(token:object) {
+    localStorage.setItem('token', JSON.stringify(token))
+  }
+
+function* auth(action: AuthAttempt) {
+    try {
+        const response = yield call(post, '/junior/register', action.payload);
+        yield call(setAuthToken, response.access_token);
+        yield put({ type: authTypes.AUTH_SUCCESS,  payload: response.access_token });
+        yield put(push('/'));
+
+    } catch (error) {
+        yield put({ type: authTypes.AUTH_FAIL });
+    }
+}
+
+function* requestLink(action: LinkRequest) {
+    try {
+        const response = yield call(post, '/junior/reset', action.payload);
```

Comment:
yeah, this function is not even fully implemented yet, cause it's blocked by SMS task. Doesn't break anything, but should I delete it for now?

Reply:
As it doesn't interfere with functionality I would advise leaving it in for now to prevent later repetition. Maybe leave a comment about it stating that it is currently unsure and help for a future implementation though.